### PR TITLE
Package Init

### PR DESCRIPTION
**Pull request by talekhinezh**

### CODE REVIEW

File: assets/account/src/lib.rs [hunk 1]
```diff
@@ -139,3 +139,5 @@ blueprint! {
         }
     }
 }
+
+package_init!(blueprint::Account::describe());
```
Comment (iamyulong) [inline assets/account/src/lib.rs:143]: Can we the macro automatically append `::describe()`?

File: radix-engine/src/model/package.rs [hunk 2]
```diff
@@ -1,18 +1,130 @@
...
+            RuntimeValue::I32(ptr) => {
+                let len: u32 = memory
+                    .get_value(ptr as u32)
+                    .map_err(|_| WasmValidationError::InvalidPackageInit)?;
+
+                // SECURITY: meter before allocating memory
+                let mut data = vec![0u8; len as usize];
+                memory
+                    .get_into((ptr + 4) as u32, &mut data)
+                    .map_err(|_| WasmValidationError::InvalidPackageInit)?;
```
Comment (iamyulong) [inline radix-engine/src/model/package.rs:70]: Minor: another place for direct memory read?

File: scrypto/src/macros.rs [hunk 3]
```diff
@@ -153,3 +153,28 @@ macro_rules! include_package {
...
     };
 }
+
+/// Exports the given blueprints into this package.
+#[macro_export]
+macro_rules! package_init {
+  ($($blueprint: expr),*) => (
+         #[no_mangle]
+          pub extern "C" fn package_init() -> *mut u8 {
+              let mut blueprints: HashMap<String, ::sbor::Type> = HashMap::new();
```
Comment (iamyulong) [inline scrypto/src/macros.rs:163]: May need to specify the canonical path for `String` to support no_std

**Review by iamyulong — APPROVED** [review]
LGTM

File: radix-engine/src/model/package.rs (hunk 2 above)
Comment (talekhinezh) [reply radix-engine/src/model/package.rs:70]: yup definitely

**Pull request merged**